### PR TITLE
Feature/ap5-25

### PR DESCRIPTION
**Pull request by jeremy-roussel**

### CODE REVIEW

File: rigex-front-end/src/components/createWellModal/createWell.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+      }
+    }, [dispatch, loading, error, data])
+
+    const [validName, setValidName] = useState(false)
+
+    const re = /^-?[0-9]{1,3}(?:\.([0-9]{1,10})?)?$/
+    const handleChange = (event) => {
+
+        let name = event.target.name
+        console.log(name)
```
Comment (BarretteBanner) [inline rigex-front-end/src/components/createWellModal/createWell.js]: 👀

File: rigex-front-end/src/components/createWellModal/createWell.js [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+        // prevent typing non-integer values or decimal into non-name fields
+        if (event.target.name !== "name"){
+            if (event.target.value === '' || re.test(event.target.value)) {
+                console.log('not name')
```
Comment (BarretteBanner) [inline rigex-front-end/src/components/createWellModal/createWell.js]: 👀

File: rigex-front-end/src/components/createWellModal/createWell.js [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+                setWellInfo((wellInfo) => {
+                   return  {
+                        ...wellInfo,
+                        [event.target.name] : event.target.value
+                    }
+                })
+            } 
+        } else if (event.target.name === "name") {
+            console.log('inside name')
```
Comment (BarretteBanner) [inline rigex-front-end/src/components/createWellModal/createWell.js]: 👀

File: rigex-front-end/src/components/createWellModal/createWell.js [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+
+    const [validName, setValidName] = useState(false)
+
+    const re = /^-?[0-9]{1,3}(?:\.([0-9]{1,10})?)?$/
+    const handleChange = (event) => {
+
+        let name = event.target.name
+        console.log(name)
+
+        // prevent typing non-integer values or decimal into non-name fields
```
Comment (BarretteBanner) [inline rigex-front-end/src/components/createWellModal/createWell.js]: One way to think of code is that it should be readable. Now just cause it's not in english doesn't mean we shouldn't be able to read code the same as we read a book. Your code should be written in a way that is clear enough that you do not need to tell other programmers what your code does in english. This violates the dry principle.

File: rigex-front-end/src/components/createWellModal/createWell.js [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+
+    useEffect(() => {
+      if (data) {
+          dispatch(wellActions.upsertWell(data.upsertWell))
+      }
+    }, [dispatch, loading, error, data])
+
+    const [validName, setValidName] = useState(false)
+
+    const re = /^-?[0-9]{1,3}(?:\.([0-9]{1,10})?)?$/
```
Comment (BarretteBanner) [inline rigex-front-end/src/components/createWellModal/createWell.js]: please no abbreviations

File: rigex-front-end/src/components/createWellModal/createWell.js [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+                    <button
+                        disabled={!validName}
+                        className={(!validName) ? "inactive" : ""}
+                        onClick={handleSubmit}
+                    >
+                        Create & add another
+                    </button>
+                    <button
+                        disabled={!validName}
+                        className={(!validName) ? "inactive" : "create"}
```
Comment (BarretteBanner) [inline rigex-front-end/src/components/createWellModal/createWell.js:118]: nicely done here

File: rigex-front-end/src/components/createWellModal/createWell.js [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+            }
+        }
+    }
+
+    return (
+        <Dialog onClose={handleWellModalClose} aria-labelledby="customized-dialog-title" open={wellModalOpen}>
+            <div className="modal-container">
+                <div className="dialog-title">
+                    Create well
+                    <button aria-label="close" className="" onClick={handleWellModalClose}>
```
Comment (BarretteBanner) [inline rigex-front-end/src/components/createWellModal/createWell.js]: this class is empty

File: rigex-front-end/src/components/createWellModal/wellModalStyles.scss [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+.dialog-title {
+    display: flex;
+    justify-content: space-between;
+    align-items: center;
+    width: 350px;
+    font-size: 1.6em;
+    margin-bottom: 15px;
+}
+
+.dialog-actions {
```
Comment (BarretteBanner) [inline rigex-front-end/src/components/createWellModal/wellModalStyles.scss]: If I traced your code correctly it looks like modal-container is the parent of all the rest of the classes therefore classes such as dialog-actions should be nested under it

File: rigex-front-end/src/pages/Rig/Rig.js [hunk 9]
```diff
@@ -1,12 +1,23 @@
...
 
 import './rig.scss';
 import rigImage from '../../assets/rig.svg';
+import CreateWellModal from '../../components/createWellModal/createWell';
 
 const Rig = ({ listOfRigs }) => {
   const { id } = useParams();
+  const [wellModalOpen, setWellModalOpen] = useState(false)
+
+  const handleWellModalClose = () => {
```
Comment (BarretteBanner) [inline rigex-front-end/src/pages/Rig/Rig.js]: I feel like this and the below method can be combined and take an argument?

**Review by BarretteBanner — APPROVED** [review]

File: rigex-front-end/src/components/createWellModal/createWell.js [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+          dispatch(wellActions.upsertWell(data.upsertWell))
+      }
+    }, [dispatch, loading, error, data])
+
+    const [validName, setValidName] = useState(false)
+
+    const handleChange = (event) => {
+
+        if (event.target.name !== "name"){
+            if (event.target.value === '' || /^-?[0-9]{1,3}(?:\.([0-9]{1,10})?)?$/.test(event.target.value)) {
```
Comment (mmevans) [inline rigex-front-end/src/components/createWellModal/createWell.js]: Could you assign this regex to an expression so we know what its testing for?

File: rigex-front-end/src/pages/Rig/Rig.js [hunk 11]
```diff
@@ -50,20 +52,24 @@ const Rig = ({ listOfRigs, handleRigModalOpen }) => {
...
           <div className="text-container">
             <span className="add-wells-text">
-              Let's create some wells on this rig!
+                Let's create some wells on this rig!
             </span>
-            <span className="small-text">
-              {' '}
-              Get started by creating a well.{' '}
+            <span className="small-text" >
+                {' '}Get started by creating a well.{' '}
```
Comment (mmevans) [inline rigex-front-end/src/pages/Rig/Rig.js]: if you need to add a space in html, you can use `&nbsp;`.

**Review by mmevans — APPROVED** [review]

**Pull request merged**